app/src/hid.c

```diff
@@ -82,15 +93,63 @@ int zmk_hid_unregister_mods(zmk_mod_flags_t modifiers) {
     return ret;
 }
 
+#if IS_ENABLED(CONFIG_ZMK_USB_BOOT)
+
+static zmk_hid_boot_report_t error_report = {
+    .modifiers = 0,
+    ._reserved = 0,
+    .keys = {
+        HID_ERROR_ROLLOVER, HID_ERROR_ROLLOVER, HID_ERROR_ROLLOVER,
+        HID_ERROR_ROLLOVER, HID_ERROR_ROLLOVER, HID_ERROR_ROLLOVER
+    }
+};
+
+#define HID_CHECK_ROLLOVER_ERROR() \
+    if (keys_held > HID_BOOT_KEY_LEN) { \
+        error_report.modifiers = keyboard_report.body.modifiers; \
+        return &error_report; \
+    }
+
+#endif
+
 #if IS_ENABLED(CONFIG_ZMK_HID_REPORT_TYPE_NKRO)
 
 #define TOGGLE_KEYBOARD(code, val) WRITE_BIT(keyboard_report.body.keys[code / 8], code % 8, val)
 
+#if IS_ENABLED(CONFIG_ZMK_USB_BOOT)
+static zmk_hid_boot_report_t *get_boot_report() {
+    HID_CHECK_ROLLOVER_ERROR();
+
+    boot_report.modifiers = keyboard_report.body.modifiers;
+    memset(&boot_report.keys, 0, HID_BOOT_KEY_LEN);
```

Comment:
This will break if `keys` is changed to not be one byte per element.
```suggestion
    memset(&boot_report.keys, 0, sizeof(boot_report.keys));
```